### ISSUE TITLE
System.IO.Pipelines issue ALSO on GCP (gcloud)

### ISSUE THREAD

**Issue by samgithub** [StackExchange/StackExchange.Redis#1689]
Could not load file or assembly '**System.IO.Pipelines, Version=5.0.0.0,** Culture=neutral, PublicKeyToken=cc7b13ffcd2ddd51'. 
The located assembly's manifest definition does not match the assembly reference. (0x80131040) 

I saw the issue related to **Azure** function (https://github.com/StackExchange/StackExchange.Redis/issues/1655)

This same error is occurring in **GCP - gcloud**

I have StackExchange.Redis version 2.2.4

<PackageReference Include="System.IO.Pipelines" Version="5.0.0" />  

//System.IO.Pipelines v5.0.1 doesn't work either-- 
the gcp app is not using  a cloud function
(Strangely, the app works in my local dev environment)

**Comment by samgithub** [comment]
Note: my project file has:

<PackageReference Include="System.IO.Pipelines" Version="5.0.0" />

**Comment by samgithub** [comment]
PackageReference Include="System.IO.Pipelines" Version="5.0.0"

**Comment by mgravell** [comment]
This sounds like assembly binding redirects:

https://nickcraver.com/blog/2020/02/11/binding-redirects/

**Comment by samgithub** [comment]
It turns out I had different versions of System.IO.Pipelines in my solution.  The main project  was using 5.0.0 and the dependencies ones were using 5.0.1.

I resolved it using the following steps:

1) I uninstalled all of System.IO.Pipelines and reinstalled just 5.0.0.
2) Published the app (dotnet publish -c Release -o publish -r linux-x64)
3) Checked the System.IO.Pipelines.dll in my publish folder to make sure it was version 5.0.0

Created my docker image and pushed to GCP.

It all works now.  Thanks for your help, @mgravell

I consider this Resolved for GCP.

Sam
